volmdlr/wires.py

```diff
@@ -2770,6 +2782,32 @@ def get_closing_point(self, polygon2_2d, primitive, ax=None):
 
         return closing_point
 
+    def validate_sewing_polygons(self, polygon2_2d):
+        primitive1 = self.line_segments[0]
+        closing_point = self.get_closing_point(polygon2_2d,
+                                               primitive1)
+        if closing_point != volmdlr.O2D:
+            return True
+        return False
+
+    def get_valid_sewing_polygons(self, polygon2_2d):
+        for i, primitive1 in enumerate(self.line_segments):
+            closing_point = self.get_closing_point(polygon2_2d,
+                                                          primitive1)
+            if closing_point != volmdlr.O2D:
+                if i == 0:
+                    return self
+                new_polygon_primitives = \
+                    self.line_segments[self.line_segments.index(primitive1):] + \
+                    self.line_segments[:self.line_segments.index(primitive1)]
+                new_points = []
+                for prim in new_polygon_primitives:
+                    if prim.start not in new_points:
+                        new_points.append(prim.start)
+                    if prim.end not in new_points:
+                        new_points.append(prim.end)
+                return ClosedPolygon2D(new_points)
```

Comment:
It could happen, yes. But for our application, if it does happen, it would mean a problem with the two polygons in use. Basically it shouldn't happen. Do you suggest adding a NotImplementedError or something to take that into account?

Reply:
Yes, raising a NotImplementedError or returning None could be a good idea, in case this happens on day.